### DISCUSSION TITLE
.NET Stryker - "since" flag doesn't seem to be doing anything

### DISCUSSION THREAD

**Discussion by theagilepilot** [stryker-mutator/stryker-net#2803] [Q&A] [answered]
Hello

When I run "dotnet stryker --since:<commitish> (and here I am inputting the hash of the last commit), dotnet stryker doesn't seem to only mutate the changed code. Instead it still mutates everything. Am I doing something wrong?

I want to only do mutation tests on code that I changed, for example the diff that appears between the latest commit and the commit before it.

Source: https://stryker-mutator.io/docs/stryker-net/configuration/#since-flag-committish

**Comment by rouke-broersma** [comment, marked as answer]
Please share the logs, they are most likely specifying why this is happening.

**Reply by theagilepilot** [reply]
![image](https://github.com/stryker-mutator/stryker-net/assets/70933144/1a3f0bbe-21d3-42f6-a780-aa2ac75b3912)
As you can see, only two files changed - both are test files, yet it is still finding mutants for the whole solution (there is 470 by default with the configuration I have).

**Reply by theagilepilot** [reply]
I believe I see the issue now. It's because "a non c-sharp file" in the test project was changed as the log says. I removed the json change and now it works fine. Thanks for getting me to look carefully at the logs though.

**Reply by theagilepilot** [reply]
For future reference is there a way to bypass this so it ignores non c-sharp changes when using --since?

**Reply by rouke-broersma** [reply]
You can use since ignore changes in
